src/main/java/com/google/sps/servlets/Search.java

```diff
@@ -0,0 +1,97 @@
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.gson.Gson;
+import com.google.sps.data.Recipe;
+import com.google.sps.data.UserQuery;
+import com.google.sps.util.SearchUtils;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** 
+ * Handles requests for a Datastore search based on a query string with filters to use.
+ **/
+@WebServlet("/search")
+public class Search extends HttpServlet {
+  /**
+   * Hnadles the get request for Datastore search results. Processes the url search parameters
+   * and returns a list in JSON format with all the recipe information to be rendered client side.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Initializes a user query object which will seperate the query data for ease of access.
+    UserQuery uQuery = new UserQuery(request.getParameter("query"));
+    
+    // Creates the new Query and adds the filter to be used in Datastore
+    // The filter is currently only set to find an exact match name in Datastore
+    // TODO: Composite Query Filter will be needed to combine and add all filter to query in future
+    Query query = new Query("PublicRecipe");
+    query.setFilter(SearchUtils.getRecipeNameFilter(uQuery.getName()));
+
+    // Makes the query to the datastore and converts it to a list so it can operated on.
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery datastoreResult = datastore.prepare(query);
+    List<Entity> recipes = datastoreResult.asList(FetchOptions.Builder.withLimit(10));
+
+    // Create a list of recipe entity information to be sent to the client, and then iterate through
+    // the datastore response to create the recipe objects and add them to the list
+    ArrayList<Recipe> clientRecipeInfo = new ArrayList<Recipe>();
+    for (int i = 0; i < recipes.size(); i++) {
+      Entity datastoreRecipe = recipes.get(i);
```

Comment:
if you're only using the index i to get a recipe, can you use a for each loop instead? ie:
`for (Entity datastoreRecipe : recipes)`

Reply:
Ack and fixed. I don't know why I had originally thought this wasn't possible